src/shared/condition.c

```diff
@@ -958,6 +960,118 @@ static int condition_test_file_is_executable(Condition *c, char **env) {
+        assert(IN_SET(c->type, CONDITION_MEMORY_PRESSURE, CONDITION_CPU_PRESSURE));
+
+        if (!is_pressure_supported())
+                return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Pressure Stall Information (PSI) is not supported");
+
+        p = c->parameter;
+        r = extract_many_words(&p, ":", 0, &first, &second, NULL);
+        if (r <= 0)
+                return log_debug_errno(r ?: SYNTHETIC_ERRNO(EINVAL), "Failed to parse condition parameter %s: %m", c->parameter);
+        /* If only one parameter is passed, then we look at the global system pressure rather than a specific cgroup. */
+        if (r == 1) {
+                pressure_path = path_join("/proc/pressure", c->type == CONDITION_CPU_PRESSURE ? "cpu" : "memory");
+                if (!pressure_path)
+                        return -ENOMEM;
+
+                value = first;
+        } else {
+                CGroupMask mask, required_mask = c->type == CONDITION_CPU_PRESSURE ? CGROUP_MASK_CPU : CGROUP_MASK_MEMORY;
+                const char *controller = c->type == CONDITION_CPU_PRESSURE ? "cpu.pressure" : "memory.pressure";
+                _cleanup_free_ char *slice_path = NULL;
+
+                r = cg_all_unified();
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to determine whether the unified cgroups hierarchy is used: %m");
+                if (r == 0)
+                        return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Requires the unified cgroups hierarchy");
+
+                r = cg_mask_supported(&mask);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to get supported cgroup controllers: %m");
+
+                if (!FLAGS_SET(mask, required_mask))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
+                                               "Cgroup %s controller not available.",
+                                               required_mask == CGROUP_MASK_CPU ? "CPU" : "memory");
+
+                r = cg_slice_to_path(first, &slice_path);
+                if (r < 0)
+                        return log_debug_errno(r, "Cannot determine slice \"%s\" cgroup path: %m", first);
+
+                r = cg_get_path(SYSTEMD_CGROUP_CONTROLLER, slice_path, controller, &pressure_path);
+                if (r < 0)
+                        return log_debug_errno(r, "Error getting cgroup pressure path from %s: %m", slice_path);
+
+                value = second;
+        }
+
+        /* If a value including the average type and comparison are passed, parse them,
+         * otherwise we assume just a plain percentage that will be checked if it is
+         * strictly smaller than the current pressure average over 5 minutes. */
+        if (startswith(value, "avg")) {
+                _cleanup_free_ char *key = NULL;
+                const char *e = value;
+
+                r = extract_first_word(&e, &key, "!<=>", EXTRACT_RETAIN_SEPARATORS);
```

Comment:
wht's the usecase for allowing all the operators here? i mean it's very unlikely anyone wants to statr more services only when under pressure? so it sounds we can just drop this?

I'd really dumb this down I must say...

Reply:
Well I considered it, but given it comes practically for free, and it's even probably simpler to support it rather than enforce the opposite, so why not? I'm sure someone will find a use for it at some point :-)

Reply:
There's big benefit in keeping things simple. I'd not add infra here that has no usecase, and needs extra docs.

Let's be a bit frugal with features: only add what really is useful — we can always extend things later if needed.